book/running-wallaroo/resilience-crash.md

```diff
@@ -0,0 +1,176 @@
+
+Since recovery involves a rollback to the last successful checkpoint, any data that was processed _after_ that checkpoint will have to be resent by the sources.
+
+## Crash Recovery in Practice
+
+When a crashed worker is restarted, if it can find its resilience files in the path specified by `--resilience-dir`, it will automatically start the recovery process. The simplest way to do this is to rerun the worker using the same command it was originally run with.
+
+For example, if we were running the bundled Python example [word_count_with_dynamic_keys](https://github.com/WallarooLabs/wallaroo/tree/master/examples/python/word_count_with_dynamic_keys/), this might look something like:
+
+### Running the crash resilience example
+
+In order to run the example you will need Machida with resilience enabled, Giles Sender, Data Receiver, and the Cluster Shutdown tool. If you haven't yet set up wallaroo, please visit our [setup](https://docs.wallaroolabs.com/book/getting-started/choosing-an-installation-option.html) instructions to get started.
+
+#### Starting new shells
+
+You will need 4 shells to run this example.
+For each shell you're expected to setup, you'd have to run the following to configure the Wallaroo environment:
+
+```bash
+cd ~/wallaroo-tutorial/wallaroo-{{ book.wallaroo_version }}
+source bin/activate
+cd examples/python/word_count_with_dynamic_keys
+```
+
+1. Create the path where the workers will save their resilience snapshots.
+
+        mkdir -p /tmp/resilience-dir
+
+2. Shell 1: Start the Metrics UI
+
+    To start the Metrics UI run:
+
+    ```bash
+    metrics_reporter_ui start
+    ```
+
+    You can verify it started up correctly by visiting [http://localhost:4000](http://localhost:4000).
+
+    If you need to restart the UI, run:
+
+    ```bash
+    metrics_reporter_ui restart
+    ```
+
+    When it's time to stop the UI, run:
+
+    ```bash
+    metrics_reporter_ui stop
+    ```
+
+    If you need to start the UI after stopping it, run:
+
+    ```bash
+    metrics_reporter_ui start
+    ```
+
+3. Shell 1: Start a data receiver
+
+    ```bash
+    data_receiver --listen 127.0.0.1:7002 | tee received.txt
+    ```
+
+4. Shell 2: Start initializer (remember to use `machida3-resilience` if using Python 3)
+    ```bash
+    machida-resilience --application-module word_count_with_dynamic_keys \
+      --in 127.0.0.1:7010 --out 127.0.0.1:7002 \
+      --metrics 127.0.0.1:5001 --control 127.0.0.1:6000 \
+      --data 127.0.0.1:6001 --external 127.0.0.1:5050 \
+      --name initializer --cluster-initializer --worker-count 2 \
+      --resilience-dir /tmp/resilience-dir \
+      --ponythreads=1 --ponynoblock --run-with-resilience \
```

Comment:
My preference is that we keep all of the Pony-specific command line args at the end of the command, so in this case I think we should move `--run-with-resilience`  so that it comes before `--ponythreads=1 --ponynoblock`.

Reply:
This example & the others that you comment on are all part of the original `resilience.md` doc.  @aturley, should I fix 'em all up now or punt to a later PR?